pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -719,6 +722,24 @@ func (nc *Controller) monitorNodeHealth() error {
 			}
 			if observedReadyCondition.Status == v1.ConditionUnknown {
 				if nc.useTaintBasedEvictions {
+					// If useTaintBasedEvictions is true, we would not evict pods
+					// But we still want to mark Pod.Status.Reason as "NodeLost"
+					selector := fields.OneTermEqualSelector(core.PodHostField, node.Name).String()
+					options := metav1.ListOptions{FieldSelector: selector}
+					pods, err := nc.kubeClient.CoreV1().Pods(metav1.NamespaceAll).List(options)
+					if err != nil {
+						return err
+					}
+					for _, pod := range pods.Items {
+						// Defensive check
+						if pod.Spec.NodeName != node.Name {
+							continue
```

Comment:
why the defensive check, we dont trust the selector would do its job correctly ?

Reply:
Just in case...